setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -43,7 +49,14 @@ public function addExcludePatterns(array $excludePatterns)
      */
     public function getList($path)
     {
+
         $realPath = realpath($path);
+        $isGeneration = strpos($realPath, DIRECTORY_SEPARATOR . 'generation' . DIRECTORY_SEPARATOR) !== false;
```

Comment:
Is it possible to restrict that check to the relative path below the Magento root. Just in case, somebody installs Magento below a directory called "generation"? It's unlikely but you never know...